src/core/tiledscene/qgstiledscenelayerrenderer.cpp

```diff
@@ -75,51 +90,550 @@ bool QgsTiledSceneLayerRenderer::render()
     mElapsedTimer.start();
   }
 
+  mSceneToMapTransform = QgsCoordinateTransform( mSceneCrs, rc->coordinateTransform().destinationCrs(), rc->transformContext() );
+
   mRenderer->startRender( context );
+  const bool result = renderTiles( context );
+  mRenderer->stopRender( context );
+  mReadyToCompose = true;
+
+  return result;
+}
+
+void QgsTiledSceneLayerRenderer::setLayerRenderingTimeHint( int time )
+{
+  mRenderTimeHint = time;
+}
+
+QgsTiledSceneRequest QgsTiledSceneLayerRenderer::createBaseRequest()
+{
+  const QgsRenderContext *context = renderContext();
+  const QgsRectangle mapExtent = context->mapExtent();
+
+  // calculate maximum screen error in METERS
+  const double maximumErrorPixels = context->convertToPainterUnits( mRenderer->maximumScreenError(), mRenderer->maximumScreenErrorUnit() );
+  // calculate width in meters across the middle of the map
+  const double mapYCenter = 0.5 * ( mapExtent.yMinimum() + mapExtent.yMaximum() );
+  const double mapWidthMeters = context->distanceArea().measureLine(
+                                  QgsPointXY( mapExtent.xMinimum(), mapYCenter ),
+                                  QgsPointXY( mapExtent.xMaximum(), mapYCenter )
+                                );
+  const double mapMetersPerPixel = mapWidthMeters / context->outputSize().width();
+  const double maximumErrorInMeters = maximumErrorPixels * mapMetersPerPixel;
```

Comment:
> that's assuming that MUPP is in meters, if not, multiply by a factor

That's the issue I ran into -- because we set scene layers to a geographic crs, if you just load in a scene layer initially then your project will be in geographic units, and there's no meaningful map unit -> meter conversion we can do. 

We could potentially bypass this with some (more!) logic to only use QgsDistanceArea when the map units are geographic... but that's just throwing more code in here :laughing: